pkg/R/deserialize.R

```diff
@@ -122,21 +123,24 @@ readRowList <- function(obj) {
   # the numCols bytes inside the read function in order to correctly
   # deserialize the row.
   rawObj <- rawConnection(obj, "r+")
-  numCols <- SparkR:::readInt(rawObj)
-  rowOut <- SparkR:::readRow(rawObj, numCols)
-  close(rawObj)
-  rowOut
+  on.exit(close(rawObj))
```

Comment:
A slightly better way to address this would be to pass in `rawObj` as an argument to `readRowList` and close it after the function returns. (This is only used in utils.R)